integration_test.go

```diff
@@ -140,6 +147,24 @@ func findDeployById(deployId string, deploys []*Deploy) bool {
 	return false
 }
 
+func findDeployPort(deployId string, deploys []*Deploy) int {
```

Comment:
Change to `findDeployById(id string, deploys []*Deploy) *Deploy` (With a missing deploy returning nil)

Then the tests can be changed to assert if returned deploy is not nil